src/main/java/com/microsoft/graph/httpcore/middlewareoption/GraphClientOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.graph.httpcore.middlewareoption;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import javax.annotation.Nonnull;
+
+import com.microsoft.graph.CoreConstants;
+
+/**
+ * Options to be passed to the telemetry middleware.
+ */
+public class GraphClientOptions implements IMiddlewareControl{
```

Comment:
use the request option interface from kiota instead and remove that interface from core to avoid duplication.